ports/unix/coverage.c

```diff
@@ -150,6 +150,43 @@ static MP_DEFINE_CONST_OBJ_TYPE(
 static const mp_obj_str_t str_no_hash_obj = {{&mp_type_str}, 0, 10, (const byte *)"0123456789"};
 static const mp_obj_str_t bytes_no_hash_obj = {{&mp_type_bytes}, 0, 10, (const byte *)"0123456789"};
 
+#if MICROPY_ENABLE_FINALISER && MICROPY_TRACKED_ALLOC
+// Dummy class which contains a tracked allocation freed from its finaliser
```

Comment:
I realised after writing this custom coverage class & test that the other way to get coverage would be to build the `coverage` variant with `MICROPY_MBEDTLS_CONFIG_BARE_METAL`, so the new unit test covers the tracked allocation changes.

This might actually be a better approach, as I think this config doesn't disable anything apart from calling libc functions... What do you think, @dpgeorge?

Reply:
If enabling `MICROPY_MBEDTLS_CONFIG_BARE_METAL` on the unix port (coverage variant) is possible, then yes that seems like a good approach.  In general less code in this `coverage.c` file is better (although it's unavoidable at times), and also it's better to emulate bare-metal for testing where possible, because then this unix-coverage build becomes a good proxy for coverage testing of bare-metal code.

Reply:
It all seems to pass here, have just pushed it to let the coverage analysis have the final word...

Reply:
Needed some tweaking, but seems to be working well. As a bonus it found a bug in my original implementation (the cleanup was running on soft reset `gc_sweep_all` and triggering double frees as the tracked nodes were already freed in this case!)